### PR TITLE
Add a CMake macro for creating Python bindings

### PR DESCRIPTION
**Pull request by luator**
## Description

Add a CMake macro for creating Python bindings.

This is a macro that is already used in a few packages (e.g. robot_interfaces) but always defined locally.  I think it is generic enough that it makes sense to put it here, so any package that uses pybind11_catkin has it available.

Usage Example (assuming `srcpy/py_my_bindings.cpp` exisits in the project):

    add_python_modul(py_my_bindings.cpp)

## How I Tested

By using it in robot_interfaces and verifying that a working Python module is generated.

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [ ] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: cmake/macros.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+# Create a Python module using pybind11
+#
+# Args:
+#     module_name:  The name of the module.  This macro expects the
+#         implementation to be found in a file `srcpy/${module_name}.cpp`.
+#     ARGN:  Any additional arguments are added to `target_link_libraries` so
+#         use this for any additional libraries that are not already covered by
+#         `${catkin_LIBRARIES}`.
+macro(add_python_module module_name)
```
Comment (MaximilienNaveau) [inline cmake/macros.cmake]: I would say this name is too general. Something including pybind11 and catkin would be appreciated.

Comment (luator) [reply cmake/macros.cmake]: Good point.  Not sure what's the best order to stack the names.  I went for `add_catkin_pybind11_module` now.

Comment (MaximilienNaveau) [reply cmake/macros.cmake]: All good, I used [this](https://github.com/machines-in-motion/mpi_cmake_modules/blob/master/cmake/pkg-config.cmake#L830) in mpi_cmake_module

So I may go for `catkin_add_pybind11_module`

Comment (luator) [reply cmake/macros.cmake]: Not sure I understand your comment correctly, shall I rename it?  I'm also good with `catkin_add_pybind11_module`.

Comment (luator) [reply cmake/macros.cmake]: I renamed it to `catkin_add_pybind11_module` now.

Comment (MaximilienNaveau) [reply cmake/macros.cmake]: Perfect good to merge! Sorry I was confusing

**Review by MaximilienNaveau — APPROVED** [review]

**Pull request merged**